tests/unit/features/RandomFourierDotFeatures_unittest.cc

```diff
@@ -176,3 +176,58 @@ TEST(RandomFourierDotFeatures, add_to_dense_test)
 
 }
 
+TEST(RandomFourierDotFeatures, gauss_coeff_dim)
+{
+	int32_t num_dims = 50;
+	int32_t num_vecs = 5;
+	int32_t D = 100;
+
+	float64_t mat[] = {
```

Comment:
what is this? use sgmatrix, and set it to const values. **The dimensions of the projection dont depend on the content**

Reply:
Right, all it needs is the num_dims of the matrix to construct the random coefficients matrix.

Reply:
my point is we dont need any random data in here. That just adds noise to the code, and is not needed